### ISSUE TITLE
[bug]: KMZ File Upload Not Recognized When Specified as MIME Type

### ISSUE THREAD

**Issue by oliverox** [pingdotgg/uploadthing#949]
### Provide environment information

```bash
System:
    OS: macOS 14.6.1
    CPU: (8) arm64 Apple M1 Pro
    Memory: 1.99 GB / 16.00 GB
    Shell: 3.3.1 - /usr/local/bin/fish
  Binaries:
    Node: 20.11.1 - ~/.nvm/versions/node/v20.11.1/bin/node
    npm: 10.2.4 - ~/.nvm/versions/node/v20.11.1/bin/npm
    pnpm: 9.10.0 - ~/Library/pnpm/pnpm
  Browsers:
    Brave Browser: 128.1.69.162
    Chrome: 128.0.6613.137
    Safari: 17.6
    Safari Technology Preview: 18.0
  npmPackages:
    @uploadthing/react: ^7.0.1 => 7.0.1
    typescript: ^5.6.2 => 5.6.2
    uploadthing: ^7.0.1 => 7.0.1
```

### Describe the bug

When attempting to upload a file with the .kmz extension, the file is not being picked up by UT even if the correct **application/vnd.google-earth.kmz+xml** MIME type is specified. Moreover, when the file picker dialog shows up, the other non *.kmz files are not disabled. This issue occurs even though other file types work as expected with their corresponding MIME types.

### Link to reproduction

https://stackblitz.com/edit/github-kaem89

### To reproduce

Click on "Choose File" to select a .kml file to upload.
=> Nothing happens.

However, changing the MIME type to "blob" allows the .kml (or any file) to be uploaded successfully.

### Additional information

Bug is reproducible on both latest and legacy versions of UT.
https://x.com/oliveroxenham/status/1834803745894670599

### 👨‍👧‍👦 Contributing

- [ ] 🙋‍♂️ Yes, I'd be down to file a PR fixing this bug!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by markflorkowski** [comment]
Can you provide sample `.kmz` and `.kml` files I can test with?

**Comment by oliverox** [comment]
I sure can. I am currently travelling but will share as soon as I get some internet access.

**Comment by oliverox** [comment]
You can get a KML file here: https://utfs.io/f/5Bp1JPFX6nSx9Vh6JEMecTo7vnCiflpF1Q8WNHYE3J4DKS6u

**Comment by juliusmarminge** [comment]
it appears my computed doesn't recognize the mime-type here:

![CleanShot 2024-09-16 at 10 33 05](https://github.com/user-attachments/assets/cd1cf5a6-7b0e-4549-8174-ad4f5c644494)

**Comment by markflorkowski** [comment]
Yeah, confirmed on mimtype.io, seems like browser does not detect.

![image](https://github.com/user-attachments/assets/6e761390-d3ed-484b-bd45-330284a3849e)

We could potentially build in functionality to the components to handle cases like these, though we wouldn't be able to get the filtering in the file selection window, just pre-upload validation.
